Sources/Client/Client/Client+WebSocket.swift

```diff
@@ -47,30 +47,48 @@ extension Client {
         let webSocketOptions = stayConnectedInBackground ? WebSocketOptions.stayConnectedInBackground : []
         let webSocketProvider = defaultWebSocketProviderType.init(request: request, callbackQueue: callbackQueue)
         
-        return WebSocket(webSocketProvider, options: webSocketOptions, logger: logger) { [unowned self] event in
-            guard case .connectionChanged(let connectionState) = event else {
-                if case .notificationMutesUpdated(let user, _, _) = event {
-                    self.userAtomic.set(user)
-                    return
-                }
-                
-                self.updateUserUnreadCount(event: event) // User unread counts should be updated before channels unread counts.
-                self.updateChannelsForWatcherAndUnreadCount(event: event)
-                return
-            }
-            
+        return WebSocket(webSocketProvider, options: webSocketOptions, logger: logger, onEvent: handleWebSocketEvent)
```

Comment:
This creates a retain cycle if I'm not mistaken.

Reply:
Fixed